custom/logistics/tasks.py

```diff
@@ -101,3 +113,98 @@ def add_products_to_loc(api):
     endpoint = api.endpoint
     synchronization(None, endpoint.get_locations, api.location_sync, None, None, 100, 0,
                     filters={"is_active": True})
+
+
+def sync_stock_transactions(domain, endpoint, facilities, checkpoint, date, limit=100, offset=0):
+    # todo: should figure out whether there's a better thing to be doing than faking this global form
+    try:
+        xform = XFormInstance.get(docid='ilsgateway-xform')
+    except ResourceNotFound:
+        xform = XFormInstance(_id='ilsgateway-xform')
+        xform.save()
+    for facility in facilities:
+        sync_stock_transactions_for_facility(domain, endpoint, facility, xform, checkpoint, date, limit, offset)
+        offset = 0  # reset offset for each facility, is only set in the context of a checkpoint resume
+
+
+def sync_stock_transactions_for_facility(domain, endpoint, facility, xform, checkpoint,
+                           date, limit=100, offset=0):
+    """
+    Syncs stock data from StockTransaction objects in ILSGateway to StockTransaction objects in HQ
+    """
+    has_next = True
+    next_url = ""
+    section_id = 'stock'
+    supply_point = facility
+    case = get_supply_point_by_external_id(domain, supply_point)
+    if case:
```

Comment:
small, but we could just return if there's not a case. that way there's not an extra tab on the majority of the function